### PR TITLE
Rewrote inspect-value as a multimethod so developers can write methods for it. 

### PR DESCRIPTION
**Pull request by pdenno**
I rewrote inspect-value as a multimethod so that developers can write methods for it. inspect is already multimethods, but inspect-value was an ordinary function. I have a print-method on a type I defined. With this patch, it uses it, even in collections displayed in the inspector.

### CODE REVIEW

**Comment by bbatsov** [comment]
Makes sense I guess. A few remarks:

* The commit message should not end with a `.`
* There's a bit of trailing whitespace in your commit
* You should probably add a test or two covering the addition of new print functions

File: src/cider/nrepl/middleware/util/inspect.clj [hunk 1]
```diff
@@ -370,3 +386,5 @@
 (defn inspect-print [x]
   (doseq [component (:rendered (inspect-render (fresh) x))]
     (inspect-print-component component)))
+
```
Comment (Malabarba) [inline src/cider/nrepl/middleware/util/inspect.clj]: What are these extra lines?

Comment (pdenno) [inline src/cider/nrepl/middleware/util/inspect.clj]: A mistake.

**Comment by bbatsov** [comment]
@pdenno Ping :-)

**Comment by pdenno** [comment]
I wrote some tests, got rid of the extra lines at the end of the file.

**Comment by bbatsov** [comment]
Looks good. Squash the two commits together and we're good to go.

**Comment by pdenno** [comment]
Done, I think!

**Comment by bbatsov** [comment]
I still see 3 commits. You have to use an interactive rebase - see this for details https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

**Comment by pdenno** [comment]
Thanks. I'll try again.

**Comment by bbatsov** [comment]
@pdenno ping :-)

**Comment by pdenno** [comment]
I could not fix this pull request; I started a new one. Sorry for the problems this may have caused.

**Pull request closed without merging**